third_party/proton/proton/profile.py

```diff
@@ -8,12 +9,29 @@
 DEFAULT_PROFILE_NAME = "proton"
 
 
+def _is_cuda():
```

Comment:
Any reason not folding these two functions into their callsite in `_select_backend`? They are only used there for now. That way we can query the backend once.